apps/api/api-cf.yml

```diff
@@ -97,8 +102,10 @@ Resources:
       Stage: $default
 
   ApiRole:
-    Type: AWS::IAM::Role
+    Type: !If [UseJPLRoles, Custom::JplRole, AWS::IAM::Role]
```

Comment:
It appears using conditionals for `Type:`  is very much not supported, and you end up with:
```
Template format error: [/Resources/Role] Every Type member must be a string.
```

which cfn-lint correctly warns about:
```
E3001 Type has to be a string at 
```

My guess is we'll have to do something like:
* write two roles: `ApiRole` and `JplApiRole`
* conditionally deploy those roles
* conditionally reference those roles

Reply:
I'm considering using jinja to inject the `Type`, since I'm worried about copy/paste errors in the `ApiRole` vs `JplApiRole` approach. That would effectively make *every* cloudformation template a jinja template, though, and doing through static analysis would mean rendering/validate for both platforms.

Reply:
I'm about 1/4 the way through doing my dual role suggestion above, and I think I agree we should jinja-ize it. This is *a lot* of copy-paste just to implement a if condition for `Type`. 

We already render before static analysis, so I think that's okay -- we can do a matrix of the security environments for the static analysis. 

It'll probably be more flexible in the long run (David mentioned deploying to CalTech as well today) but oof, this is all much bigger than I want though.